proposals/NNNN-disallow-inout-explicit-convs.md

```diff
@@ -0,0 +1,112 @@
+---
+title: "NNNN - Disallow all explicit conversions in inout parameters"
+params:
+  authors:
+    - spall: Sarah Spall
+  sponsors:
+    - spall: Sarah Spall
+  status: Under Consideration
+---
+
+* Issues: https://github.com/microsoft/hlsl-specs/issues/639
+
+## Introduction
+
+DXC inconsistently allows some C style explicit casts to produce LValues and be
+used as arguments for inout parameters.  However, C style explicit casts do not
+produce LValues and inout parameters require LValues.
```

Comment:
Consider adding that HLSL should mimic C++ behavior as closely as possible, so because C++ doesn't allow LValues as a result of these casts, HLSL shouldn't either.